### ISSUE TITLE
Select : keyboard interaction of live demo differs from accessibility description

### ISSUE THREAD

**Issue by Regine-Lambrecht** [carbon-design-system/carbon-website#3690]
The [Accessibility tab of the Select component](https://carbondesignsystem.com/components/select/accessibility) says it should open the option list with Up/down arrow, and then only _navigate_ options in the listbox when using Up/down arrow keys again. ENTER should select the item. It also says that ESC closes without changing the selected option. This is also what W3C recommends for the [single select combobox in aria](https://www.w3.org/WAI/ARIA/apg/patterns/combobox/examples/combobox-select-only/), explaining that it improves the HTML SELECT behaviour: "the following differences in behavior are implemented to improve both accessibility and general usability".

The issue is that the [live demo of the SELECT in CARBON DS](https://carbondesignsystem.com/components/select/usage#live-demo) does not behave like that.

So how do we implement the keyboard interaction described in the accessibility tab ?

**Comment by tw15egan** [comment]
@Regine-Lambrecht This is the behavior I am seeing, what interaction is not matching?

This link may be easier for testing: https://react.carbondesignsystem.com/iframe.html?args=&id=components-select--default&viewMode=story

**Comment by tw15egan** [comment]
Wondering if this is a Windows-specific behavior? Do other `<select>` elements behave this way on your machine? We are just rendering the browser/os native `select` element, and do not implement any extra keystroke functionality. 


 I'm only able to test on Mac at the moment, but this is what I am getting. 

![2023-08-09 14 47 12](https://github.com/carbon-design-system/carbon-website/assets/11928039/680798e4-f1c8-48f8-850c-73bdf5d49f23)

**Comment by Regine-Lambrecht** [comment]
Yes, the SELECT behaves differently on Windows, apparently ;( I found documentation of that lack of support. 
You can see it [here](https://www.magentaa11y.com/checklist-web/select/)

There has been a [study ](https://www.24a11y.com/2019/select-your-poison/) that still recommends using the SELECT, but it does not consider the usability issue of having to memorize the currently selected option and come back to it if you want to escape the list of options without modifying. That is the same issue as with radio buttons: you cannot discover the options without selecting each one you just wanted to hear, and you have to come back to the first if this is finally what you wanted. You also lose track of any default selection once you discovered the other options. But ok, there is no alternative for radio buttons, and they all behave like that until the W3C created a custom one based on ARIA that could behave better :).

For the select feature, all other clickable components that also display a list of options with unique selection after user interaction, do have a "browse" only interaction, like the SELECT on Mac:

- an [action button with focus](https://www.w3.org/WAI/ARIA/apg/patterns/menu-button/examples/menu-button-actions/), 
- an [action button with descendant ](https://www.w3.org/WAI/ARIA/apg/patterns/menu-button/examples/menu-button-actions-active-descendant/), 
- a [navigation menu](https://www.w3.org/WAI/ARIA/apg/patterns/menu-button/examples/menu-button-links/) with sub-menu (you expand, your browse the options, you click ENTER to select)
- [editor menu bar](https://www.w3.org/WAI/ARIA/apg/patterns/menubar/examples/menubar-editor/)
- all navigation menus of links "behind" a clickable entry

We have to remember that the visitor doesn't know which HTML is behind the scene. He just sees/hear something that is clickable and displays a list of choices. Why not coding them so that they all behave the same: arrow down opens, other arrow down puts focus/reads aloud without selecting, ENTER selects, ESC closes without changing the initial value ?

To get that keyboard interaction for MacOS AND Windows, one should use the ARIA [Select-only Combobox pattern](https://www.w3.org/WAI/ARIA/apg/patterns/combobox/examples/combobox-select-only/):
See also https://github.com/microsoft/sonder-ui/tree/master/src/components/select

**Comment by tw15egan** [comment]
@Regine-Lambrecht I would recommend trying the `Dropdown` component then, as that should implement the select-only combobox pattern shown here. The `Select` component is intentionally left untouched and is simply the browser's native implementation without any added styling (except the unopened display)

**Comment by guidari** [comment]
We are planning the OS handle the Select component, since it is not a issue in Carbon.
If you like a consitent behavior you can try the Dropdown component.
